### ISSUE TITLE
indent is made too dependent of tabSize which forbid some settings

### ISSUE THREAD

**Issue by perror** [James-Yu/LaTeX-Workshop#4019]
## Pre-checks
> _Please change the following `[ ]` to `[x]` for confirmation._
- [x ] The feature request has not been suggested in this repository (as far I've seen in the previous issue reports).

## The Missed
> _Is your feature request related to a problem? Please provide a clear and concise description of what the problem is._
- I tried to tweak the `latexindent` script to fit my requisite which was to indent a block with two spaces but I could not achieve this without setting the `tabSize` to 2 as well (I want my `tabSize` to be set to 8 as it is the default in many settings).

## The Solution*
> _Please provide a solution you would like to have._
- In `latexformatter.js` I would like to replace the following code:
```javascript
...
        const options = vscode.window.activeTextEditor.options;
        const tabSize = options.tabSize ? +options.tabSize : 4;
        const useSpaces = options.insertSpaces;
        const indent = useSpaces ? ' '.repeat(tabSize) : '\\t';
...
```
By this code:
```javascript
...
        const options = vscode.window.activeTextEditor.options;
        const indent = options.indent
...
```
With an option `indent` that is set to `\t` by default.

## Anything Else?
> _Add any other context about the feature request below._
- LaTeX-Workshop is really useful, so thanks for all your work !

**Comment by jlelong** [comment]
Why don't you just change the value of `tabSize` in LaTeX mode?

```json
    "[latex]": {
        "editor.tabSize": 2
    }
```

**Comment by perror** [comment]
Because `tabSize` is 8, not 2. What you suggest is just a (quick and dirty) workaround not a fix.

**Comment by James-Yu** [comment]
I don’t understand your use case. You want 2 spaces for latexindent and eight when you manually press tab in a latex document? Sorry but I don’t get it.

**Comment by perror** [comment]
Well, the indentation of a code base is usually counted in number of spaces. And, it is usually recommended to use spaces to indent code's blocks (for portability reasons).

Some editors, as VSCode and others, have changed this by considering "tabs only indented code" where you can resize the tabulation character at will to make it fit your need. 

But, the problem is that the tabulation character is usually represented as 8 spaces in numerous platforms (e.g. Gitlab or Github). So, if you set your `tabSize` to other value 8, you will have a very different code when displayed on these platforms (or in other developers settings).

In VSCode, the core code and most of the extensions allow the two approaches : 
-  Either you indent with tabulations and you resize the tabulation as you want.
- Or you indent with spaces and you leave tabulation to 8 (default).

Somehow, your code break this implicit rule that allow both ways of doing, because you linked `tabSize` and `indent` deep down in your code. More precisely, the problem is in this line:
```javascript
        const indent = useSpaces ? ' '.repeat(tabSize) : '\\t';
```
In fact, `indent` and `tabSize` shouldn't be linked like that. `indent` may be set as a tabulation but it should also be possible to set it as a number of spaces that is not correlated to `tabSize`. And, here you break the rule.

I hope I was a bit more explicit this time... Sorry if it is still not the case.

**Comment by perror** [comment]
Yes, I can hard-code it inside the arguments of `latexindent`, this will provide a workaround (but not a fix). Okay for that solution, thanks.

**Comment by James-Yu** [comment]
This is the fix. A default configuration cannot work for every use case.

**Comment by perror** [comment]
Hi,
Do not reopen this issue, I am just adding the work-around I found for my problem.

After several trials I finally managed to stay with `tabSize` at 8 and get indent to 2 spaces by adding the following to my local settings:
```
  "latex-workshop.latexindent.args": [
    "-c",
    "%DIR%/",
    "%TMPFILE%",
    "-y=defaultIndent: '  '"
  ]
```
Forcing the `defaultIndent`directly to `latexindent` allow to bypass the value of `latex-workshop.indent` which solve my problem.

Thanks for the code, this is very useful.

**Comment by perror** [comment]
Hi,
Do not reopen this issue, I just wanted to explain the work-around I found for my problem in case other people hit a similar problem.

So, after several trials and errors I finally added this user setting to my `settings.js`file:
```js
  "latex-workshop.latexindent.args": [
    "-c",
    "%DIR%/",
    "%TMPFILE%",
    "-y=defaultIndent:'  '"
  ]
```
By setting directly the indent in the `latexindent` command line I bypass totally the `indent` which allow me to stay with `tabSize` to 8.

Anyway, thanks for your code, this is very helpful!
